crypto/rand/randfile.c

```diff
@@ -280,8 +280,16 @@ const char *RAND_file_name(char *buf, size_t size)
     struct stat sb;
 #endif
 
+    /* Find the directory where the state file should go. */
+#ifdef OPENSSL_SYS_WINDOWS
+    if ((s = getenv("RANDFILE")) == NULL) {
+        s = getenv("TEMP");
+        if (s == NULL)
+            s = getenv("APPDATA");
```

Comment:
By default it is CryptGenRandom (you only get BCryptGenRandom if _WIN32_WINNT is explicitly defined at Configure time). That happens in RAND_poll(). If an error occurs in RAND_poll() then no entropy gets added to the pool. That's ok because when you try and get some RAND_bytes it will fail because not enough entropy is available...unless of course you've additionally seeded it with bad data from a shared random file. In which case it will all apparently work but your seed is bad.